src/components/UserDetails/UserDetails.jsx

```diff
@@ -0,0 +1,238 @@
+            <FormLabel className={styles['user-details-form-label']}>First Name</FormLabel>
+            <Input
+              id="first-name"
+              placeholder="Enter first name"
+              {...register('firstName')}
+              isRequired
+              isDisabled={!canEditForm}
+            />
+            <Box className={styles['error-box']}>{errors.firstName?.message}</Box>
+          </FormControl>
+          <FormControl isRequired>
+            <FormLabel className={styles['user-details-form-label']}>Last Name</FormLabel>
+            <Input
+              id="last-name"
+              placeholder="Enter last name"
+              {...register('lastName')}
+              isRequired
+              isDisabled={!canEditForm}
+            />
+            <Box className={styles['error-box']}>{errors.lastName?.message}</Box>
+          </FormControl>
+          <FormControl isRequired>
+            <FormLabel className={styles['user-details-form-label']}>Phone Number</FormLabel>
+            <Input
+              type="tel"
+              id="phone-number"
+              placeholder="Enter phone number"
+              {...register('phoneNumber')}
+              isRequired
+              isDisabled={!canEditForm}
+            />
+            <Box className={styles['error-box']}>{errors.phoneNumber?.message}</Box>
+          </FormControl>
+          <FormLabel className={styles['user-details-form-label']}>Email</FormLabel>
+          <Input
+            type="email"
+            id="email"
+            placeholder="Enter email"
+            value={user.email}
+            {...register('email')}
+            isRequired
+            isDisabled
+          />
+          <Box className={styles['error-box']}>{errors.email?.message}</Box>
+          <FormControl isRequired={getValues('newPassword')?.length > 0}>
+            <FormLabel className={styles['user-details-form-label']}>New Password</FormLabel>
+            <Input
+              type="password"
+              id="new-password"
+              placeholder="Enter new password"
+              {...register('newPassword')}
+              isRequired={getValues('newPassword')?.length > 0}
+              isDisabled={!canEditForm}
+            />
+            <Box className={styles['error-box']}>{errors.password?.message}</Box>
+          </FormControl>
+          <FormControl isRequired={getValues('newPassword')?.length > 0}>
+            <FormLabel className={styles['user-details-form-label']}>
+              Re-enter New Password
+            </FormLabel>
+            <Input
+              type="password"
+              id="confirm-password"
+              placeholder="Re-enter new password"
+              {...register('confirmPassword')}
+              isRequired={getValues('newPassword')?.length > 0}
+              isDisabled={!canEditForm}
+            />
+            <Box className={styles['error-box']}>{errors.confirmPassword?.message}</Box>
+          </FormControl>
```

Comment:
I can't seem to submit the form without changing the password as well? To recreate: I tried changing my phone number. Clicking save moves the cursor to highlight the new password field, and changes to the phone number is not made.